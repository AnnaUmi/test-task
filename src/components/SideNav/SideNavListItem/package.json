{
  "name": "side-nav-list-item",
  "version": "0.0.0",
  "private": true,
  "main": "./SideNavListItem.js"
}
