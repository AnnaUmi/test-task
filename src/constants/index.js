import PropTypes from 'prop-types';

/* eslint-disable import/prefer-default-export */
export const NORMALIZE_URL =
  'https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css';

export const SIDE_NAV_ITEM_TYPES = {
  ListItem: 'ListItem',
  SubMenu: 'SubMenu',
};

export const routerProps = PropTypes.shape({
  pathName: PropTypes.string,
  route: PropTypes.string,
  query: PropTypes.object,
  back: PropTypes.func,
  beforePopState: PropTypes.func,
  prefetch: PropTypes.func,
  push: PropTypes.func,
  reload: PropTypes.func,
  replace: PropTypes.func,
});

export const sideNavSubMenuItemProps = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  url: PropTypes.string,
};

export const sideNavItemProps = {
  ...sideNavSubMenuItemProps,
  type: PropTypes.oneOf([SIDE_NAV_ITEM_TYPES.ListItem, SIDE_NAV_ITEM_TYPES.SubMenu]).isRequired,
};

export const noop = () => {};
