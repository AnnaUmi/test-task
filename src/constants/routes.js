import { SIDE_NAV_ITEM_TYPES } from '.';

export const HEADER_NAV_ROUTES = [
  {
    route: '/',
    label: 'PRISM Logo',
    id: '/',
  },
  {
    route: '/brand',
    label: 'Brand',
    id: '/brand',
  },
  {
    route: '/product',
    label: 'Product',
    id: '/product',
  },
  {
    route: '/marketing',
    label: 'Marketing',
    id: '/marketing',
  },
  {
    route: '/components',
    label: 'Components',
    id: '/components',
  },
];

const componentsSideNav = [
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Buttons',
    id: 'components/buttons',
    url: '/components/buttons',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Typography',
    id: 'components/typography',
    url: '/components/typography',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Grid',
    id: 'components/grid',
    url: '/components/grid',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Spacing',
    id: 'components/spacing',
    url: '/components/spacing',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Colors',
    id: 'components/colors',
    url: '/components/colors',
  },
  {
    type: SIDE_NAV_ITEM_TYPES.ListItem,
    label: 'Breadcrumbs',
    id: 'components/breadcrumbs',
    url: '/components/breadcrumbs',
  },
];

export const SIDE_NAV_ROUTES = {
  '/components': componentsSideNav,
};
