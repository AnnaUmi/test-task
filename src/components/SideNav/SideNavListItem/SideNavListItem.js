import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

import { sideNavItemProps } from '../../../constants';
import { SideNavListItemStyled } from '../SideNav.styles';

const SideNavListItem = ({ item, isActive }) => (
  <Link href={item.url}>
    <SideNavListItemStyled isActive={isActive}>{item.label}</SideNavListItemStyled>
  </Link>
);

SideNavListItem.propTypes = {
  item: PropTypes.shape(sideNavItemProps).isRequired,
  isActive: PropTypes.bool.isRequired,
};

export default SideNavListItem;
