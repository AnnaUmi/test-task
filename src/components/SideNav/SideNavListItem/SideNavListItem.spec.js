import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import SideNavListItem from './SideNavListItem';
import { SIDE_NAV_ITEM_TYPES } from '../../../constants';

const item = {
  label: 'test label',
  url: 'test/url',
  id: 'test/url',
  type: SIDE_NAV_ITEM_TYPES.ListItem,
};

describe('SideNavListItem', () => {
  it('should match snapshot when not active', () => {
    const tree = renderer.create(<SideNavListItem item={item} isActive={false} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when is active', () => {
    const tree = renderer.create(<SideNavListItem item={item} isActive />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
