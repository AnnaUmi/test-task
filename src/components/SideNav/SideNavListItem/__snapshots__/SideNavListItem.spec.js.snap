// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavListItem should match snapshot when is active 1`] = `
.c0 {
  padding: 20px 30px;
  font-size: 20px;
  cursor: pointer;
  background-color: #7e7e7e;
  font-weight: bold;
}

.c0:hover {
  background-color: #bdbdbd;
}

<div
  className="c0"
  onClick={[Function]}
>
  test label
</div>
`;

exports[`SideNavListItem should match snapshot when not active 1`] = `
.c0 {
  padding: 20px 30px;
  font-size: 20px;
  cursor: pointer;
}

.c0:hover {
  background-color: #bdbdbd;
}

<div
  className="c0"
  onClick={[Function]}
>
  test label
</div>
`;
